src/qt/sendcoinsdialog.cpp

```diff
@@ -789,9 +789,7 @@ void SendCoinsDialog::coinControlChangeEdited(const QString& text)
         }
         else // Valid address
         {
-            CKeyID keyid;
-            addr.GetKeyID(keyid);
-            if (!model->havePrivKey(keyid)) // Unknown change address
+            if (model->isMine(addr.Get()) != ISMINE_SPENDABLE) // Unknown change address
```

Comment:
if you feel like it, you might add a `const CTxDestination dest = addr.Get();` before that line and then use `if (!bool(model->IsMine(dest) & ISMINE_SPENDABLE))`.

Though, it shouldn't make a difference.